internal/client/job.go

```diff
@@ -55,19 +45,78 @@ func (c *Project) doJob(ctx context.Context, job *pb.Job, ui terminal.UI) (*pb.J
 	return c.doJobMonitored(ctx, job, ui, nil)
 }
 
+// setupLocalJobSystem does the pre-work required to run jobs locally. This includes:
+// - figure out if jobs should be executed locally or remotely.
+// - if job should be executed locally, start a local runner
+// - FUTURE: if a job should be executed remotely, but local VCS is present and dirty, warn.
+// This lives separately from DoJob because the logs and exec commands need to conditionally warm up the
+// local job infrastructure, but don't actually create a job (the server does).
+func (c *Project) setupLocalJobSystem(ctx context.Context) (isLocal bool, newCtx context.Context, err error) {
+	// Automatically determine if we should use a local or a remote runner
+	newCtx = ctx
+
+	if c.useLocalRunner == nil {
```

Comment:
Would love a comment above here reminding me what this tri-state being `nil` implies. i.e. "nil means we should auto-detect" or something.